pkg/resources/gce/gce.go

```diff
@@ -1037,6 +1041,105 @@ func deleteServiceAccount(cloud fi.Cloud, r *resources.Resource) error {
 	return err
 }
 
+// containsOnlyListedIGMs returns true if all the given backend service's backends
+// are contained in the provided list of IGM resources.
+func containsOnlyListedIGMs(svc *compute.BackendService, igms []*resources.Resource) bool {
+	for _, be := range svc.Backends {
+		listed := false
+		for _, igm := range igms {
+			if strings.HasSuffix(be.Group, "/"+igm.Name) {
```

Comment:
Nit: is this matching strict enough?  Maybe add a TODO to double-check / revisit if we can make it stricter.

Reply:
Added note!